# Projeto de predição com churn

- projeto pessoal integrando análise de dados e ciência de dados (modelagem e previsão)

In [1]:
from google.colab import files
import pandas as pd

upload_de_arquivos = files.upload()

df = pd.read_csv('Churn_Modelling.csv')

df.head(10)

Saving Churn_Modelling.csv to Churn_Modelling.csv


,RowNumber,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
0,1,15634602,Hargrave,619,France,Female,42,2,0.00,1,1,1,101348.88,1
1,2,15647311,Hill,608,Spain,Female,41,1,83807.86,1,0,1,112542.58,0
2,3,15619304,Onio,502,France,Female,42,8,159660.80,3,1,0,113931.57,1
3,4,15701354,Boni,699,France,Female,39,1,0.00,2,0,0,93826.63,0
4,5,15737888,Mitchell,850,Spain,Female,43,2,125510.82,1,1,1,79084.10,0
5,6,15574012,Chu,645,Spain,Male,44,8,113755.78,2,1,0,149756.71,1
6,7,15592531,Bartlett,822,France,Male,50,7,0.00,2,1,1,10062.80,0
7,8,15656148,Obinna,376,Germany,Female,29,4,115046.74,4,1,0,119346.88,1
8,9,15792365,He,501,France,Male,44,4,142051.07,2,0,1,74940.50,0
9,10,15592389,H?,684,France,Male,27,2,134603.88,1,1,1,71725.73,0


In [2]:
df.shape

(10000, 14)

In [6]:
df.columns

Index(['RowNumber', 'CustomerId', 'Surname', 'CreditScore', 'Geography',
       'Gender', 'Age', 'Tenure', 'Balance', 'NumOfProducts', 'HasCrCard',
       'IsActiveMember', 'EstimatedSalary', 'Exited'],
      dtype='object')

Através do comando acima foi possível notar a dimensão do dataset. Ele está agrupado por 10.000 linhas e 14 colunas.

Com "df.head(10)" foi possível enxergar quais colunas o dataset traz para participação de informações relevantes. As colunas são:
- RowNumber
- CustomerId
- Surname
- CreditScore
- Geography
- Gender
- Age
- Tenure
- Balance
- NumOfProducts
- HasCrCard
- IsActiveMember
- EstimatedSalary
- Exited

## Verificação de dados duplicados

In [3]:
df.duplicated()

,0
0,False
1,False
2,False
3,False
4,False
...,...
9995,False
9996,False
9997,False
9998,False


## Verificação de dados nulos

In [4]:
df.isnull()  #ou df.isna()

,RowNumber,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9995,False,False,False,False,False,False,False,False,False,False,False,False,False,False
9996,False,False,False,False,False,False,False,False,False,False,False,False,False,False
9997,False,False,False,False,False,False,False,False,False,False,False,False,False,False
9998,False,False,False,False,False,False,False,False,False,False,False,False,False,False


Aparentemente, o comando acima gerou linhas com valores indicadores de "False" para dados nulos, significa que as colunas estão preenchidas

In [5]:
df.isnull().sum()

,0
RowNumber,0
CustomerId,0
Surname,0
CreditScore,0
Geography,0
Gender,0
Age,0
Tenure,0
Balance,0
NumOfProducts,0


A soma de todos os valores nulos em cada coluna foi igual a 0, reforçando, assim, que a base de dados não possui valores nulos.

## Verificação de valores nas colunas

In [17]:
df.dtypes

,0
RowNumber,int64
CustomerId,int64
Surname,object
CreditScore,int64
Geography,object
Gender,object
Age,int64
Tenure,int64
Balance,float64
NumOfProducts,int64


In [13]:
df[['CustomerId', 'Surname','Gender']].head(12)

,CustomerId,Surname,Gender
0,15634602,Hargrave,Female
1,15647311,Hill,Female
2,15619304,Onio,Female
3,15701354,Boni,Female
4,15737888,Mitchell,Female
5,15574012,Chu,Male
6,15592531,Bartlett,Male
7,15656148,Obinna,Female
8,15792365,He,Male
9,15592389,H?,Male


In [14]:
df[['Surname', 'IsActiveMember']].head(12)

,Surname,IsActiveMember
0,Hargrave,1
1,Hill,1
2,Onio,0
3,Boni,0
4,Mitchell,1
5,Chu,0
6,Bartlett,1
7,Obinna,0
8,He,1
9,H?,1


In [15]:
df[['Surname', 'Exited']].head(12)

,Surname,Exited
0,Hargrave,1
1,Hill,0
2,Onio,1
3,Boni,0
4,Mitchell,0
5,Chu,1
6,Bartlett,0
7,Obinna,1
8,He,0
9,H?,0


## Início à Análise de dados

### Verificando a estimativa de salário por país/região

In [18]:
df.groupby("Geography")["EstimatedSalary"].sum()

,EstimatedSalary
Geography,
France,5.008945e+08
Germany,2.536936e+08
Spain,2.463143e+08


### Taxa de churn

In [19]:
df['Exited'].mean()

np.float64(0.2037)

### Pessoas que saíram por País/Região

In [21]:
df.groupby("Geography")["Exited"].mean().sort_values(ascending=False) * 100

,Exited
Geography,
Germany,32.443204
Spain,16.673395
France,16.154767


- 32.44% dos clientes cancelaram na Alemanha.
- 16.67% cancelaram na Espanha.
- 16.15% cancelaram na França.

## Churn por gênero

In [23]:
df.groupby("Gender")["Exited"].mean() * 100

,Exited
Gender,
Female,25.071539
Male,16.455928


## Idade por Churn

In [24]:
df.groupby("Exited")["Age"].mean()

,Age
Exited,
0,37.408389
1,44.837997


## Número de produtos contratados por Churn

In [25]:
df.groupby("NumOfProducts")["Exited"].mean()

,Exited
NumOfProducts,
1,0.277144
2,0.075817
3,0.827068
4,1.000000


## Saldo vs Churn

In [26]:
df.groupby("Exited")["Balance"].mean()

,Balance
Exited,
0,72745.296779
1,91108.539337


## Atividades e cartão de crédito

In [30]:
df.groupby("IsActiveMember")["Exited"].mean() * 100

,Exited
IsActiveMember,
0,26.850897
1,14.269074


In [31]:
df.groupby("HasCrCard")["Exited"].mean() * 100

,Exited
HasCrCard,
0,20.814941
1,20.184266


## Correlação

In [33]:
df.corr(numeric_only=True)

,RowNumber,CustomerId,CreditScore,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
RowNumber,1.000000,0.004202,0.005840,0.000783,-0.006495,-0.009067,0.007246,0.000599,0.012044,-0.005988,-0.016571
CustomerId,0.004202,1.000000,0.005308,0.009497,-0.014883,-0.012419,0.016972,-0.014025,0.001665,0.015271,-0.006248
CreditScore,0.005840,0.005308,1.000000,-0.003965,0.000842,0.006268,0.012238,-0.005458,0.025651,-0.001384,-0.027094
Age,0.000783,0.009497,-0.003965,1.000000,-0.009997,0.028308,-0.030680,-0.011721,0.085472,-0.007201,0.285323
Tenure,-0.006495,-0.014883,0.000842,-0.009997,1.000000,-0.012254,0.013444,0.022583,-0.028362,0.007784,-0.014001
Balance,-0.009067,-0.012419,0.006268,0.028308,-0.012254,1.000000,-0.304180,-0.014858,-0.010084,0.012797,0.118533
NumOfProducts,0.007246,0.016972,0.012238,-0.030680,0.013444,-0.304180,1.000000,0.003183,0.009612,0.014204,-0.047820
HasCrCard,0.000599,-0.014025,-0.005458,-0.011721,0.022583,-0.014858,0.003183,1.000000,-0.011866,-0.009933,-0.007138
IsActiveMember,0.012044,0.001665,0.025651,0.085472,-0.028362,-0.010084,0.009612,-0.011866,1.000000,-0.011421,-0.156128
EstimatedSalary,-0.005988,0.015271,-0.001384,-0.007201,0.007784,0.012797,0.014204,-0.009933,-0.011421,1.000000,0.012097
